pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1053,32 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+func resourceAttributes(proxy *Proxy) *otlpcommon.KeyValueList {
+	// TODO(zirain): generate resource attributes for VM
+	if proxy.IsVM() {
+		return nil
+	}
+
+	return &otlpcommon.KeyValueList{
+		Values: []*otlpcommon.KeyValue{
+			{
+				Key:   "k8s.cluster.name",
+				Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: proxy.Metadata.ClusterID.String()}},
+			},
+			{
+				Key: "k8s.service.name",
+				// this seems a little hack, but in istio it works, we do the samething for `serviceCluster` in tracing
+				Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: proxy.XdsNode.Cluster}},
+			},
+			{
+				Key:   "k8s.pod.namespace",
+				Value: &otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: proxy.Metadata.Namespace}},
+			},
+			{
+				Key:   "k8s.pod.name",
```

Comment:
what if its not a pod

Reply:
Actually workload name is more commonly Deployment anyways?not pod name

Reply:
yes, I misunderstand that
pod name is more useful, let me do some research

Reply:
We should probably use the same values we are setting in `baggage` for ambient

Reply:
should we follow the existing [OpenTelemetry Resource Semantic Conventions](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/) ?

Reply:
@kyessenov @lei-tang @costinm

Reply:
I think we should obey the OTEL rule

Reply:
It seems like we should probably align with Otel and baggage. So if those aren't already aligned maybe we should align those as well

Reply:
Agree with @hzxuzhonghu , Otel semantic conventions take priority over anything in Istio or Ambient.
`k8s.pod.name` seems like the right thing for pods, but not every sidecar runs in a pod, could be a job, for example.

Reply:
> but not every sidecar runs in a pod, could be a job, for example

good catch, seems it's better to remove `k8s.deployment.name` for now.

Reply:
reverted deployment name

@kyessenov @howardjohn further options on this PR?

Reply:
Sorry, what I meant is that deployment is actually useful, but it should only be set for deployment pods. The attribute is called `k8s.deployment.name`.

Reply:
We should, IMO, have the same logic as https://github.com/istio/istio/blob/373fc89518c986c9f48ed3cd891930da6fdc8628/pilot/pkg/networking/core/v1alpha3/listener.go#L1653. If that logic is considered wrong, I have no objections to changing it, but they should be consistent (and consistent with OTEL)

Reply:
> Sorry, what I meant is that deployment is actually useful, but it should only be set for deployment pods. The attribute is called k8s.deployment.name.

IIRC, there's no way to determine a pod came from a deployment or job, revert it this first, may add it back later.

Reply:
Does https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/ not work?

Reply:
> Does https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/ not work?

but can we get this from `model.Proxy` ?